src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -2868,7 +2872,7 @@ private bool HasImplicitReferenceTypeParameterConversion(TypeParameterSymbol sou
             }
 
             // * From T to a type parameter U, provided T depends on U.
-            if ((destination.TypeKind == TypeKind.TypeParameter) &&
+            if (destination is TypeParameterSymbol { AllowByRefLike: false } &&
```

Comment:
>if (destination is TypeParameterSymbol { AllowByRefLike: false } && [](http://example.com/codeflow?start=12&length=67)

Why is `AllowByRefLike: false` required? Isn't it sufficient to check `source.AllowByRefLike` (above) and `source.DependsOn((TypeParameterSymbol)destination)`?

Reply:
> Why is `AllowByRefLike: false` required? Isn't it sufficient to check ```source.AllowByRefLike (above)``` and ```source.DependsOn((TypeParameterSymbol)destination)```?

Why do you think it should be sufficient? Dependency between type parameters says nothing about their ```AllowByRefLike``` property (dependent type parameters can have different values for it) and reference conversions do not exist for those that have ```AllowByRefLike: true```.